test/spec/ProjectManager-test.js

```diff
@@ -120,6 +120,68 @@ define(function (require, exports, module) {
                     expect(stat.isFile).toBe(true);
                 });
             });
+            
+            it("should not display excluded entry when resolved and written to", function () {
+                var opFailed = false,
+                    doneResolving = false,
+                    doneWriting = false,
+                    doneLooking = false,
+                    found = false,
+                    entry;
+                
+                runs(function () {
+                    waitsForDone(SpecRunnerUtils.copy(tempDir + "/git/", tempDir + "/.git/"), "copy git to .git");
+                });
+                
+                runs(function () {
+                    FileSystem.resolve(ProjectManager.getProjectRoot().fullPath + ".git/", function (err, e, stat) {
+                        if (err) {
+                            opFailed = true;
+                            return;
+                        }
+                        entry = e;
+                        doneResolving = true;
+                    });
+                });
+                
+                waitsFor(function () {
+                    return !opFailed && doneResolving;
+                }, "FileSystem.resolve()", 500);
+
+                runs(function () {
+                    var file = FileSystem.getFileForPath(entry.fullPath + "test");
+                    file.write("hi there!", function (err) {
+                        if (err) {
+                            opFailed = true;
+                            return;
+                        }
+                        doneWriting = true;
+                    });
+                });
+                
+                waitsFor(function () {
+                    return !opFailed && doneWriting;
+                }, "create a file under .git", 500);
+                
+                runs(function () {
+                    ProjectManager.showInTree(entry)
```

Comment:
@peterflynn, this call is actually not rejected, so the test is wrong. do you have a suggestion on how to find out if the entry is in the tree? i'm thinking inspecting actual DOM, but may there's a better way.